### PR TITLE
D3ASIM-2741 - Folders Structure Change(If Duplicate Areas are present)

### PR DESCRIPTION
**Pull request by psurender**
This PR is on park until Spyros review and also this is dependent on other jira items to complete, just ,not to forget from my side, i am raising this PR.

### CODE REVIEW

File: src/d3a/d3a_core/export.py [hunk 1]
```diff
@@ -543,6 +557,9 @@ def _plot_ess_energy_trace(self, energy: dict, subdir: str, root_name: str):
         if len(data) == 0:
             return
         plot_dir = os.path.join(self.plot_dir, subdir)
+        # to do
+        if area.slug != "grid":
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:561]: This is not universal. The user can name their root areas whatever they want. Why do you want to omit adding the uuid for the root are in the first place?
If you realy want to do it, you could test for `area.parent == None`

File: src/d3a/d3a_core/export.py [hunk 2]
```diff
@@ -98,7 +98,7 @@ def export_json_data(self, directory: dir):
 
     @staticmethod
     def _file_path(directory: dir, slug: str):
-        file_name = ("%s.csv" % slug).replace(' ', '_')
+        file_name = ("{0}.csv".format(slug)).replace(' ', '_')
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:101]: If you want to change this, please use an fstring rather.

File: src/d3a/d3a_core/export.py [hunk 3]
```diff
@@ -204,7 +204,8 @@ def _export_area_offers_bids_csv_files(self, area, directory, file_suffix,
         (*-bids/offers/balancing-offers.csv files)
         return: dict[out_keys]
         """
-        file_path = self._file_path(directory, f"{area.slug}-{file_suffix}")
+        file_path = self._file_path(directory,
+                                    f"{area.slug}-{file_suffix}")
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:208]: There are a lot of these refatorings in this PR. Can you tell me the reason fo this?

File: src/d3a/d3a_core/export.py [hunk 4]
```diff
@@ -296,6 +299,10 @@ def _plot_device_stats(self, address_list: list, device_strategy):
         # converting address_list into plot_dir by slugifying the members
         plot_dir = os.path.join(self.plot_dir,
                                 "/".join([slugify(node).lower() for node in address_list][0:-1]))
+        # to do
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:302]: Please remove all of these comments from this PR. Thanks!

File: src/d3a/d3a_core/export.py [hunk 5]
```diff
@@ -510,16 +524,16 @@ def plot_ess_energy_trace(self, area, subdir):
...
             if child.children:
                 self.plot_ess_energy_trace(child, new_subdir)
 
-    def _plot_ess_energy_trace(self, energy: dict, subdir: str, root_name: str):
+    def _plot_ess_energy_trace(self, energy: dict, subdir: str, area):
         """
         Plots ess energy trace for each knot in the hierarchy
         """
-
+        root_name = area.slug
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:536]: Where do you need this variable?

**Review by hannesdiedrich — COMMENTED** [review]
Left some comments, will review after they are addressed

File: src/d3a/d3a_core/export.py [hunk 6]
```diff
@@ -307,7 +314,11 @@ def plot_energy_profile(self, area: Area, subdir: str):
         """
         self.endpoint_buffer.trade_profile.add_sold_bought_lists(
             self.endpoint_buffer.trade_profile.traded_energy_profile)
-        new_subdir = os.path.join(subdir, area.slug)
+        # to do
+        if area.slug != "grid":
+            new_subdir = os.path.join(subdir, area.slug + "-" + area.uuid)
+        else:
+            new_subdir = os.path.join(subdir, area.slug)
```
Comment (hannesdiedrich) [inline src/d3a/d3a_core/export.py:321]: Minor and im not sure whether this is possible, but I wonder whether you can find a smarter way to add the uuid in a more central place and not doing this in every function.

**Comment by psurender** [comment]
This PR i have raised for not to miss, This PR is kept on Hold, and also initially as i mentioned this is PR require some changes.
but this PR changes covers in the PR - https://github.com/gridsingularity/d3a/pull/866

hence closing this PR

**Pull request closed without merging**